### PR TITLE
Test app and client response improvements

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**
This PR includes improvements to the test app, and addresses some problems with different responses when `cancel` remediation values aren't included in the ION JSON. Additionally, `contextualData` and `settings` response payloads are available as options.

### CODE REVIEW

File: OktaIdxExample/OktaIdxExample/Signin/Signin.swift [hunk 1]
```diff
@@ -53,6 +53,36 @@ public class Signin {
...
+        }
+        
+        switch option.type {
+        case .identify:
+            return "Next"
+        
+        case .skip:
+            return "Skip"
+            
+        case .selectEnrollProfile: fallthrough
```
Comment (oleggnidets-okta) [inline OktaIdxExample/OktaIdxExample/Signin/Signin.swift:68]: What about `case .selectEnrollProfile, .enrollProfile`? It makes more explicit intention. 

It is tightly tied to the idea that after `selectEnrollProfile` is `entrollProfile`. I would use `fallthrough` when it doesn't matter what goes next.

File: OktaIdxExample/OktaIdxExample/Signin/TableView Cells/IDXSeparatorTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//
+//  IDXSeparatorTableViewCell.swift
+//  OktaIdxExample
+//
+//  Created by Mike Nachbaur on 2021-03-19.
```
Comment (oleggnidets-okta) [inline OktaIdxExample/OktaIdxExample/Signin/TableView Cells/IDXSeparatorTableViewCell.swift:5]: Please, update the header here and in `IDXTitleTableViewCell` :)

File: OktaIdxExample/OktaIdxExample/Signin/View Controllers/IDXRemediationTableViewController.swift [hunk 3]
```diff
@@ -58,6 +56,21 @@ class IDXRemediationTableViewController: UITableViewController, IDXResponseContr
...
     
+    var shouldShowNavigationBar: Bool {
+        guard let response = response,
+              let app = response.app,
+              let remediation = response.remediation
+        else {
+            return true
+        }
+        
+        return !app.label.isEmpty && remediation.remediationOptions.count != 0
```
Comment (oleggnidets-okta) [inline OktaIdxExample/OktaIdxExample/Signin/View Controllers/IDXRemediationTableViewController.swift:67]: What about `!remediation.remediationOptions.isEmpty`?

**Review by oleggnidets-okta — APPROVED** [review]
It looks great!

File: OktaIdxExample/OktaIdxExample/Signin/TableView Cells/IDXSeparatorTableViewCell.swift (hunk 2 above)
Comment (mikenachbaur-okta) [reply OktaIdxExample/OktaIdxExample/Signin/TableView Cells/IDXSeparatorTableViewCell.swift:5]: Hah! Thanks, I thought I got them all. That reminds me that I should add a default comment header template in the Xcode project.

File: OktaIdxExample/OktaIdxExample/Signin/Signin.swift (hunk 1 above)
Comment (mikenachbaur-okta) [reply OktaIdxExample/OktaIdxExample/Signin/Signin.swift:68]: This is for the human-readable titles, so it's less about communicating the intent behind each remediation type, but about the messaging to convey to the user.

**Pull request closed without merging**